txpipe/psf_diagnostics.py

```diff
@@ -172,6 +172,415 @@ def plot_histogram(self, function, output_name, xlabel, edges):
         return results
 
 
+class TXTauStatistics(PipelineStage):
+    """
+    Compute and plot PSF Tau statistics where the definition of Tau stats are eq.20-22
+    of Gatti et al 2023.
+    """
+    name     = "TXTauStatistics"
+    parallel = False
+    inputs   = [("shear_catalog"           , ShearCatalog),
+                ("shear_tomography_catalog", TomographyCatalog),
+                ("star_catalog"            , HDFFile),
+                ("rowe_stats"              , HDFFile),
+               ]
+
+    outputs  = [
+                ("tau0p"    , PNGFile),
+                ("tau2p"    , PNGFile),
+                ("tau5p"    , PNGFile),
+                ("tau_stats", HDFFile),
+               ]
+
+
+    config_options = {
+                       "min_sep"       : 0.5,
+                       "max_sep"       : 250.0,
+                       "nbins"         : 20,
+                       "bin_slop"      : 0.01,
+                       "sep_units"     : "arcmin",
+                       "psf_size_units": "sigma",
+                     }
+
+    def run(self):
+        import treecorr
+        import h5py
+        import matplotlib
```

Comment:
can you add `import emcee` up here - I try to do this so that all the modules that get imported happen at the start, so if there is anything missing you find out quickly.  Also then the version numbers are saved in any files output.

Reply:
Fixed. Also moved `from scipy.stats import qmc`  and `import treecorr` here as well.

Reply:
Actually, I tried moving this these imports to just below `run()` but it doesn't seem to propagate to the other functions so I get `'qmc' is not defined` and  `'treecorr' is not defined`

Reply:
Yes, I usually have to put it in both of them. It doesn't slow anything down.

Reply:
Ahh ok, I had thought you meant *move it* to just beneath run(), didn't realize you actually meant to leave it in the function as well. Fixed now.